app/src/main/java/com/lagradost/cloudstream3/extractors/Embedgram.kt

```diff
@@ -0,0 +1,37 @@
+package com.lagradost.cloudstream3.extractors
+
+import com.lagradost.cloudstream3.SubtitleFile
+import com.lagradost.cloudstream3.app
+import com.lagradost.cloudstream3.utils.ExtractorApi
+import com.lagradost.cloudstream3.utils.ExtractorLink
+import com.lagradost.cloudstream3.utils.getQualityFromName
+import com.lagradost.cloudstream3.utils.httpsify
+
+class Embedgram : ExtractorApi() {
+    override val name = "Embedgram"
+    override val mainUrl = "https://embedgram.com"
+    override val requiresReferer = true
+
+    override suspend fun getUrl(
+        url: String,
+        referer: String?,
+        subtitleCallback: (SubtitleFile) -> Unit,
+        callback: (ExtractorLink) -> Unit
+    ) {
+        val document = app.get(url, referer = referer).document
+        val link = document.select("video source:last-child").attr("src")
+        val quality = document.select("video source:last-child").attr("title")
+        callback.invoke(
+            ExtractorLink(
+                this.name,
+                this.name,
+                httpsify(link),
+                "$mainUrl/",
+                getQualityFromName(quality),
+                headers = mapOf(
+                    "Range" to "bytes=0-"
```

Comment:
🤨 does not exoplayer take care of this?

Reply:
idk, but it doesn't work without that, same case with [Egybest](https://github.com/ImZaw/cloudstream-extensions-arabic/blob/9b2fe358a2ef7c3a4946b30c9ec5f2dd1609f98b/EgyBestProvider/src/main/kotlin/com/egybest/EgyBestProvider.kt#L255-L256)